src/EditorFeatures/Core/Implementation/LineSeparators/AdornmentManager.cs

```diff
@@ -302,6 +312,33 @@ private void UpdateInvalidSpans()
             }
         }
 
+        private NormalizedSpanCollection GetCollapsedSpans()
+        {
+            var elisionSnapshot = _textView.VisualSnapshot as IProjectionSnapshot;
+            if (elisionSnapshot == null)
+            {
+                return NormalizedSpanCollection.Empty;
+            }
+
+            // Get the source spans from the visual snapshot.  The source spans will be for the 
+            // uncollapsed regions.  So we can infer the collapsed regions for the spans between
+            // the uncollapsed ones.
+
+            var sourceSpans = elisionSnapshot.GetSourceSpans();
+            var collapsedSpans = new List<Span>();
+            for (var i = 0; i < sourceSpans.Count - 1; i++)
+            {
+                var span1 = sourceSpans[i];
```

Comment:
Rather than doing fancy logic like this, can't you just get the source spans and negate your .IntersectsWith check somewhere?